arches/app/datatypes/datatypes.py

```diff
@@ -1535,7 +1535,7 @@ def append_to_document(self, document, nodevalue, nodeid, tile, provisional=Fals
             for f in tile.data[str(nodeid)]:
                 val = {"string": f["name"], "nodegroup_id": tile.nodegroup_id, "provisional": provisional}
                 document["strings"].append(val)
-        except KeyError as e:
+        except (KeyError, TypeError) as e:
```

Comment:
Maybe we don't need the KeyError any longer?

Reply:
Resolving this as-is for now, please let me know if that needs to be changed.